calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -505,3 +518,103 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
 
 	return buf.String(), nil
 }
+
+// validateMetadata compares the Metadata for a ResourceObject and warns if anything was
+// changed that will not be changed as a part of the update. Any discrepancies to the metadata
+// will be changed to match the metadata available in the stored object except for Labels and
+// Annotations.
+func validateMetadata(stored, changed ResourceObject) ResourceObject {
```

Comment:
This isn't just validate (and if we remove the warnings won't be validate at all).

I think the method name should indicate that this is merging ObjectMetadata for an Update.